clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -193,9 +197,16 @@ class CIRGenBuilderTy : public mlir::OpBuilder {
   mlir::cir::ConstantOp getNullValue(mlir::Type ty, mlir::Location loc) {
     if (ty.isa<mlir::cir::PointerType>())
       return getNullPtr(ty, loc);
-    assert(ty.isa<mlir::IntegerType>() && "NYI");
-    return create<mlir::cir::ConstantOp>(loc, ty,
-                                         mlir::IntegerAttr::get(ty, 0));
+    
+    mlir::TypedAttr attr;
+    if (ty.isa<mlir::IntegerType>())
```

Comment:
Any reason why `mlir::IntegerAttr` is this still around?

Reply:
It handles booleans. Basically, we use a built-in `mlir::BoolAttr` to hold the boolean value of constants. The underlying type of this `BoolAttr` is a built-in integer. I'll create a custom CIR bool attribute to work around this on a follow-up PR.

Reply:
NVM. Seems like you fixed it already.

Reply:
Right, I just added a `mlir::cir::BoolAttr`, so you are probably good to remove this!